core/app.cc

```diff
@@ -360,13 +363,25 @@ void application::prepare_argv(elf::program *program)
     }
 
     // Load vdso library if available
-    int auxv_parameters_count = 2;
+    int auxv_parameters_count = 4;
     _libvdso = program->get_library("libvdso.so");
     if (!_libvdso) {
         auxv_parameters_count--;
         WARN_ONCE("application::prepare_argv(): missing libvdso.so -> may prevent shared libraries specifically Golang ones from functioning\n");
     }
 
+    // Initialize random bytes array so it can be passed as AT_RANDOM auxv vector
+    if (getrandom(random_bytes, sizeof(random_bytes), GRND_RANDOM) != sizeof(random_bytes)) {
```

Comment:
Nitpick: since you anyway intend to fall back to pseudo-random number generation, I think you could have just used urandom - i.e., not pass GRND_RANDOM. Then getrandom() is guaranteed to succeed (if number of bytes is less than 256). 
But there's nothing wrong your code either.

Reply:
I think it is an excellent idea as it will simplify the code.